tests/e2e_suite.py

```diff
@@ -87,6 +91,7 @@ def run(args):
 
         # For now, if a test fails, the entire test suite if stopped
         if status is TestStatus.failure:
+            success = False
```

Comment:
Might as well raise?

Reply:
We could but we still want to display the results of the tests that have succeeded so far at the end of the run. We'd have to `try: except:` the whole test loop which I think is not as neat for now. 

I will add support for randomised test in the next PR so I'll probably re-work this file anyway.